base_layer/wallet/src/ffi.rs

```diff
@@ -157,16 +187,27 @@ pub unsafe extern "C" fn add_pending_transaction_outputs(
     pending_tx: *mut PendingTransactionOutputs,
 ) -> bool
 {
+    if wallet.is_null() {
+        return false;
+    }
+    if pending_tx.is_null() {
+        return false;
+    }
+    return true;
     // append this data to the wallet
 }
 
 /// Initialize a Transaction struct to be populated
 #[no_mangle]
 pub unsafe extern "C" fn create_transaction(
-    tx_id: c_ulonglong,    // u64
-    offset: *const c_char, // Byte[32] - PrivateKey
+    tx_id: c_ulonglong,  // u64
+    offset: *mut c_char, // Byte[32] - PrivateKey
```

Comment:
Can you explain why this one is a mut not a const like a string would be?

Reply:
I might have misinterpreted the const char from the existing FFI stuff?

Reply:
CString::from_raw(ptr: *mut c_char)